nltk/parse/malt.py

```diff
@@ -5,245 +6,492 @@
-            input_file.close()
-            self.train_from_file(input_file.name, verbose=verbose)
-        finally:
-            input_file.close()
-            os.remove(input_file.name)
-
-    def train_from_file(self, conll_file, verbose=False):
-        """
-        Train MaltParser from a file
-
-        :param conll_file: str for the filename of the training input data
-        """
-        if not self._malt_bin:
-            raise Exception("MaltParser location is not configured.  Call config_malt() first.")
-
-        # If conll_file is a ZipFilePathPointer, then we need to do some extra
-        # massaging
-        if isinstance(conll_file, ZipFilePathPointer):
-            input_file = tempfile.NamedTemporaryFile(prefix='malt_train.conll',
-                                                     dir=self.working_dir,
-                                                     delete=False)
-            try:
-                conll_str = conll_file.open().read()
-                conll_file.close()
-                input_file.write(conll_str)
-                input_file.close()
-                return self.train_from_file(input_file.name, verbose=verbose)
-            finally:
-                input_file.close()
-                os.remove(input_file.name)
-
-        cmd = ['java', '-jar', self._malt_bin, '-w', self.working_dir,
-               '-c', self.mco, '-i', conll_file, '-m', 'learn']
-
-        ret = self._execute(cmd, verbose)
-        if ret != 0:
-            raise Exception("MaltParser training (%s) "
-                            "failed with exit code %d" %
-                            (' '.join(cmd), ret))
-
-        self._trained = True
-
-    @staticmethod
-    def _execute(cmd, verbose=False):
-        output = None if verbose else subprocess.PIPE
-        p = subprocess.Popen(cmd, stdout=output, stderr=output)
-        return p.wait()
-
-
-def demo():
-    dg1 = DependencyGraph("""1    John    _    NNP   _    _    2    SUBJ    _    _
-                             2    sees    _    VB    _    _    0    ROOT    _    _
-                             3    a       _    DT    _    _    4    SPEC    _    _
-                             4    dog     _    NN    _    _    2    OBJ     _    _
-                          """)
-    dg2 = DependencyGraph("""1    John    _    NNP   _    _    2    SUBJ    _    _
-                             2    walks   _    VB    _    _    0    ROOT    _    _
-                          """)
-
-    verbose = False
-
-    maltParser = MaltParser()
-    maltParser.train([dg1,dg2], verbose=verbose)
-
-    maltParser.parse_one(['John','sees','Mary'], verbose=verbose).tree().pprint()
-    maltParser.parse_one(['a','man','runs'], verbose=verbose).tree().pprint()
-    
-    next(maltParser.tagged_parse([('John','NNP'),('sees','VB'),('Mary','NNP')], verbose)).tree().pprint()
-    
+def taggedsent_to_conll(sentences):
+	"""
+	A module to convert the a POS tagged document stream 
```

Comment:
It's a function, strictly speaking.

It's related to https://github.com/nltk/nltk/blob/develop/nltk/parse/dependencygraph.py#L434 It might worth making this function more general. A sentence contains of tuples of different length, so a correct template is picket up. Then the DependencyGraph.to_connl() uses this function.